bentoml/serve.py

```diff
@@ -285,27 +351,52 @@ def serve_production(
         port=port,
         backlog=backlog,
     )
+
+    api_server_watcher_args = [
+        "-m",
+        SCRIPT_API_SERVER,
+        bento_identifier,
+        "--bind",
+        "fd://$(circus.sockets._bento_api_server)",
+        "--runner-map",
+        json.dumps(runner_bind_map),
+        "--working-dir",
+        working_dir,
+        "--backlog",
+        f"{backlog}",
+        "--worker-id",
+        "$(CIRCUS.WID)",
+        "--prometheus-dir",
+        prometheus_dir,
+    ]
+    # Add optional SSL args if they exist
+    api_server_watcher_args.extend(
+        ["--ssl-keyfile", ssl_keyfile]
+    ) if ssl_keyfile is not None else None  # pylint: disable=W0106
+    api_server_watcher_args.extend(
+        ["--ssl-certfile", ssl_certfile]
+    ) if ssl_certfile is not None else None  # pylint: disable=W0106
+    api_server_watcher_args.extend(
+        ["--ssl-keyfile-password", ssl_keyfile_password]
+    ) if ssl_keyfile_password is not None else None  # pylint: disable=W0106
+    api_server_watcher_args.extend(
+        ["--ssl-version", str(ssl_version)]
+    ) if ssl_version is not None else None  # pylint: disable=W0106
+    api_server_watcher_args.extend(
+        ["--ssl-cert-reqs", str(ssl_cert_reqs)]
+    ) if ssl_cert_reqs is not None else None  # pylint: disable=W0106
+    api_server_watcher_args.extend(
+        ["--ssl-ca-certs", ssl_ca_certs]
+    ) if ssl_ca_certs is not None else None  # pylint: disable=W0106
+    api_server_watcher_args.extend(
+        ["--ssl-ciphers", ssl_ciphers]
+    ) if ssl_ciphers is not None else None  # pylint: disable=W0106
```

Comment:
https://github.com/encode/uvicorn/blob/b21ecabc5bf911f571e0629438315a1e5472065c/uvicorn/config.py

Uvicorn does handle None option for some option. 

IMO, a nicer way to write None check is
```
if ssl_certfile:
   args.extend(["--ssl-certfile", ssl_certfile])
if ssl_cert_reqs:
   args.extend(...)
if ssl_version:
   ...
```

Then we can pass all of the other options straight through

Reply:
Yes, this is a weird bit of code. It doesn't (shouldn't?) actually pass `None` into uvicorn - [here](https://github.com/bentoml/BentoML/pull/2886/commits/d23713ac8cbed5320d38cc00107de0fced680145#diff-4f4b8a5d43ba42f4f34b447f11659cf2b6f322abe52efb26d5bc4cbcb89fb90cR117) is where uvicorn actually gets the args if they exist. 
I wrote it this way to try compacting the code a bit, but running it through the formatter made that a moot point. I agree that it should be reformatted to the way you proposed - much easier to understand.